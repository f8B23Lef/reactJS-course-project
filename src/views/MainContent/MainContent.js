import React from 'react';
import { useSelector } from 'react-redux';
import PropTypes from 'prop-types';
import GenreToggle from '../../components/GenreToggle/GenreToggle';
import SortDropdown from '../../components/SortDropdown/SortDropdown';
import MovieCount from '../../components/MovieCount/MovieCount';
import MovieList from '../../components/MovieList/MovieList';
import { GENRES, SORT_BY } from '../../utils/constants';
import { selectMoviesCount } from '../../store/moviesSlice';
import './MainContent.scss';

export default function MainContent(props) {
  const { onMovieClick } = props;

  const movieCount = useSelector(selectMoviesCount);

  return (
    <div className='main-content'>
      <div className='top-content'>
        <GenreToggle
          genreNames={['All', ...GENRES]}
          initialGenre='All'
        />
        <SortDropdown
          options={SORT_BY}
        />
      </div>
      <MovieCount value={movieCount} />
      <MovieList onMovieClick={onMovieClick} />
    </div>
  );
}

MainContent.propTypes = {
  onMovieClick: PropTypes.func.isRequired,
};
