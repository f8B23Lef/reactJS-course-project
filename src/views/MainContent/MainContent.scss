.main-content {
  margin: 15px 35px;
}

.top-content {
  display: flex;
  justify-content: space-between;
  align-items: baseline;
  border-bottom: 2px solid #424242;
}
