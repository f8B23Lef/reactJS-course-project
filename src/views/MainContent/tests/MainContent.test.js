import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import MainContent from '../MainContent';

jest.mock(
  '../../../components/GenreToggle/GenreToggle',
  () => () => 'GenreToggleComponent',
);
jest.mock(
  '../../../components/SortDropdown/SortDropdown',
  () => () => 'SortDropdownComponent',
);
jest.mock(
  '../../../components/MovieCount/MovieCount',
  () => () => 'MovieCountComponent',
);
jest.mock(
  '../../../components/MovieList/MovieList',
  () => () => 'MovieListComponent',
);
jest.mock('react-redux', () => ({
  ...jest.requireActual('react-redux'),
  useSelector: () => jest.fn(),
}));

describe('MainContent view', () => {
  it('renders MainContent view snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <MainContent onMovieClick={jest.fn()} />
      </BrowserRouter>,
    );

    expect(asFragment(<MainContent onMovieClick={jest.fn()} />))
      .toMatchSnapshot();
  });
});
