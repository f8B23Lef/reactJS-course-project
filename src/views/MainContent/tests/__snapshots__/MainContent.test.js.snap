// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainContent view renders MainContent view snapshot 1`] = `
<DocumentFragment>
  <div
    class="main-content"
  >
    <div
      class="top-content"
    >
      GenreToggleComponentSortDropdownComponent
    </div>
    MovieCountComponentMovieListComponent
  </div>
</DocumentFragment>
`;
